src/cpp/shared/adventure/map_events.cpp

```diff
@@ -30,6 +30,11 @@ void advManager::PasswordEvent(mapCell *cell, hero *hero) {
     gpCurPlayer->barrierTentsVisited |= 1 << barrierColor;
 }
 
+void playerData::SetBarrierTentsVisited(char x) {
+	gpCurPlayer->barrierTentsVisited += 1;
```

Comment:
This is on the right track. I recommend doing the bit-op exercises. Can you tell me why this can actually un-mark a tent as visited?